src/main/java/edu/njit/jerse/automation/AsheAutomation.java

```diff
@@ -0,0 +1,141 @@
+     * the file to extract class and method information, and then for each public method,
+     * the ASHE 'run' method is invoked to apply minimization and error correction.
+     *
+     * @param javaFile        the Java file to be processed
+     * @param projectRootPath the root path of the project, used for determining the
+     *                        relative path of the Java file
+     * @throws IOException          if an I/O error occurs when opening or parsing the file
+     * @throws ExecutionException   if the ASHE framework encounters an error during execution
+     * @throws InterruptedException if the ASHE execution is interrupted
+     * @throws TimeoutException     if the ASHE execution takes longer than the allowed time
+     */
+    private static void processJavaFile(File javaFile, String projectRootPath)
+            throws IOException, ExecutionException, InterruptedException, TimeoutException {
+        LOGGER.info("Processing Java file: {}", javaFile.getAbsolutePath());
+        CompilationUnit cu = StaticJavaParser.parse(javaFile);
+
+        String targetFile = javaFile
+                .getAbsolutePath()
+                .substring(projectRootPath.length() + 1)
+                .replace(File.separatorChar, '/');
+
+        if (targetFile.startsWith("src/main/java/")) {
+            targetFile = targetFile.substring("src/main/java/".length());
+        }
+
+        String packageName = cu.getPackageDeclaration().map(NodeWithName::getNameAsString).orElse("");
+        for (TypeDeclaration<?> type : cu.getTypes()) {
+            if (type.isPublic()) {
+                String fullClassName = packageName.isEmpty() ? type.getNameAsString() : packageName + "." + type.getNameAsString();
+                for (BodyDeclaration<?> member : type.getMembers()) {
+                    if (member instanceof MethodDeclaration method) {
+                        if (method.isPublic()) {
+                            String parameters = method.getParameters().stream()
+                                    .map(p -> p.getType().asString())
+                                    .collect(Collectors.joining(", "));
+                            String targetMethod = fullClassName + "#" + method.getNameAsString() + "(" + parameters + ")";
+                            LOGGER.info("Target method: {}", targetMethod);
+                            ASHE ashe = new ASHE();
+                            ashe.run(projectRootPath, targetFile, targetMethod);
+                        }
+                    }
+                }
+            }
+        }
+        LOGGER.info("Completed processing Java file: {}", javaFile.getAbsolutePath());
+    }
+
+    /**
+     * The entry point of the application automating ASHE.
+     * Expects two command-line arguments:
+     * the path to the directory containing Java files and the root path of the project.
+     * This method initiates the automation process on the specified directory, applying
```

Comment:
"initiates the automation process on the specified directory, applying" => "applies"